### PR TITLE
Fix combine() not working with undefined values

### LINKED ISSUE #27: combine will not work when the user sends undefined
https://github.com/cyclejs/callbags/blob/01a6e613223d2cef3b3ac37cfea9639b9c9eb28a/src/combine.ts#L31

This is a thing I have just noticed when looking through the code

### PR DESCRIPTION
**Pull request by jvanbruegge**
Fixes #27

### CODE REVIEW

**Comment by jvanbruegge** [comment]
I am quite prowd of that property test I have to say 😅

File: src/combine.ts [hunk 1]
```diff
@@ -28,8 +29,18 @@ export function combineWith<T extends [...Producer<unknown>[]], U>(
...
         } else if (t === 1) {
           combined[i] = d;
-          if (combined.indexOf(undefined) === -1)
+          if (!filled) {
+            filled = true;
+            for (let i = 0; i < combined.length; i++) {
+              if (!(i in combined)) {
+                filled = false;
+                break;
+              }
```
Comment (staltz) [inline src/combine.ts]: This for loop (which will run many times) smells to me much slower than a counter variable solution.

Check the callbag-combine implementation. Apart from the obscurely named variables, the implementation itself should be easy to build.

**Review by staltz — CHANGES_REQUESTED** [review]

File: src/combine.ts (hunk 1 above)
Comment (jvanbruegge) [reply src/combine.ts]: the trick with the empty object is a good one. Did that now

**Comment by staltz** [comment]
Good! I don't know how to approve from GitHub mobile, but the implementation looks good now ( should also be smaller in kilobyte footprint ), so approved.

I don't understand the property tests, but I trust you know what you're doing, so merge away :)

**Comment by jvanbruegge** [comment]
> I don't understand the property tests

It basically generates an array of arrays with data. For every array, a subject is spawned that receives data from an `setInterval`, draining the arrays. The combined values are then divided back into the arrays so we can check that a) all items were delivered and b) the order of items is the same

**Pull request merged**